authors/apps/authentication/tests/test_registration.py

```diff
@@ -0,0 +1,262 @@
+                         "message"], "Valid email should be alphanumeric with \
+                         the @ and . symbol")
+
+    def test_invalid_password(self):
+        ''' Test that checks if the password has valid input '''
+
+        response = self.client.post(
+            'api/users/register',
+            data=json.dumps(self.valid_password),
+            format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(json.loads(response.data)[
+                         "message"], "Valid password should be more than 8 \
+                         characters, alphanumeric, have at least one capital \
+                         letter and a symbol")
+
+    def test_duplicate_username(self):
+        ''' Test that checks if the username already exists '''
+
+        self.client.post(
+            'api/users/register',
+            data=json.dumps(self.duplicate_username),
+            format='json')
+
+        response = self.client.post(
+            'api/users/register',
+            data=json.dumps(self.duplicate_username2),
+            format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(json.loads(response.data)[
+                         "message"], "Username already exists")
+
+    def test_duplicate_email(self):
+        ''' Test that checks if the email already exists '''
+
+        self.client.post(
+            'api/users/register',
+            data=json.dumps(self.duplicate_email),
+            format='json')
+
+        response = self.client.post(
+            'api/users/register',
+            data=json.dumps(self.duplicate_email2),
+            format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(json.loads(response.data)[
+                         "message"], "Email already exists")
+
+    def test_missing_data(self):
+        ''' Test that checks if the required fields exists '''
+
+        response = self.client.post(
+            'api/users/register',
+            data=json.dumps({}),
+            format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(json.loads(response.data)[
+                         "message"], "Username, email and password are \
+                         required fields")
+
+    def test_missing_username(self):
+        ''' Test that checks if the required username field exists '''
+
+        response = self.client.post(
+            'api/users/register',
+            data=json.dumps(self.missing_username),
+            format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(json.loads(response.data)[
+                         "message"], "Username is a required field")
+
+    def test_missing_email(self):
+        ''' Test that checks if the required email field exists '''
```

Comment:
Please remember to change the single quotes into doubles for uniformity with your other files.

Reply:
I will change that. Thanks for pointing it out